documentation/src/main/asciidoc/elasticsearch-integration.asciidoc

```diff
@@ -83,17 +109,137 @@ Other options for the `hibernate.search.elasticsearch.index_management_strategy`
+* if you want to retrieve the distance in a geolocation query, install and enable the `lang-groovy` plugin,
+  see <<elasticsearch-query-spatial>>
+* if you want to be able to delete by query, install the `delete-by-query` plugin
+
+=== Mapping and indexing
+
+Like in Lucene embedded mode, indexes are transparently updated when you create or update
+entities mapped to Hibernate Search.
+Simply use familiar annotations from <<search-mapping>>.
+
+The name of the index will be the lowercased name provided to `@Indexed` (non qualified class name by default).
+Hibernate Search will map the fully qualified class name to the Elasticsearch type.
+
+==== [[elasticsearch-mapping-analyzer]] Analyzers
+
+CAUTION: Analyzers are treated differently than in Lucene embedded mode.
 
-As with Lucene in embedded mode, the indexes are updated automatically when you create or update
-entities which are mapped to Hibernate Search using the same annotations already familiar from <<search-mapping>>.
+Analyzers should be defined (by name) in your Elasticsearch configuration
+instead of using `@AnalyzerDef`, we are looking into smoothing that experience.
 
-==== Running a query on an Elasticsearch mapped entity
+[source, yml]
+.Example of elasticsearch.yml defining custom analyzers
+--
+# Custom analyzer
+index.analysis:
+  analyzer.custom-analyzer:
+    type: custom
+    tokenizer: standard
+    filter: [custom-filter, lowercase]
+  filter.custom-filter:
+    type : stop
+    stopwords : [test1, close]
+--
 
-In many cases the existing way (see <<search-query>>) of running queries should work:
-we do automatically translate the most common types of Apache Lucene queries and many of the queries generated by the Hibernate Search DSL.
+More information on Elasticsearch analyzers in particular the already defined ones can be found in link:https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-analyzers.html[their documentation].
 
-On top of translating Lucene queries, you can directly create Elasticsearch queries by using either its String format or a JSON format:
+From there, you can use an analyzer by name in your entity mappings.
+
+[source,java]
+.Using analyzers
+--
+@Entity
+@Indexed(index = "tweet")
+public static class Tweet {
+
+    @Id
+    @GeneratedValue
+    private Integer id;
+
+    @Field
+    @Analyzer(definition = "english")
+    private String englishTweet;
+
+    @Field
+    @Analyzer(definition = "whitespace")
+    private String whitespaceTweet;
+
+    @Fields({
+            @Field(name = "tweetNotAnalyzed", analyze = Analyze.NO, store = Store.YES),
+            @Field(name = "tweetWithCustom", analyzer = @Analyzer(definition = "custom-analyzer") ) })
+    private String multipleTweets;
+}
+--
+
+==== Custom field bridges
+
+You can write custom field bridges and class bridges.
+Make sure to make your bridge implementation also implement the `MetadataProvidingFieldBridge` contract.
+
+TODO: is that a good example
```

Comment:
Is implementing `MetadataProvidingFieldBridge` required only for class bridges or multi-fields fieldbridges? Or also for simple bridges?